h2o-core/src/main/java/hex/Model.java

```diff
@@ -2225,23 +2226,39 @@ public boolean testJavaScoring(Frame data, Frame model_predictions, EasyPredictM
 
   public void deleteCrossValidationModels() {
     if (_output._cross_validation_models != null) {
+      Log.info("Cleaning up CV Models for " + this._key.toString());
+      int c = 0;
       for (Key k : _output._cross_validation_models) {
         Model m = DKV.getGet(k);
-        if (m!=null) m.delete(); //delete all subparts
+        if (m!=null) {
+          m.delete(); //delete all subparts
+          c++;
+        }
       }
+      Log.info(c+" models were removed");
+      _output._cross_validation_models = null;
```

Comment:
I don't think this is a good idea. You are making a local change to the output object but the update is not being propagated to DKV.

I suggest keeping output object unmodified.

Reply:
this nullify is probably not needed anymore given how output is now handled in  cv_mainModelScores below.
Although, what we were doing before was not very clean: typically when listing for CV models, the users was able to get the keys, but the models themselves had been removed from DKV. That's why I started to nullify here.

Good point about the DKV update, it's something I need to get used to: is there any best practice to avoid forgetting the update? like a transactional syntax or sth similar? e.g. key.withValue((v) -> use and auto update v)?

Reply:
fixed and closed